public/app/features/alerting/state/alertDef.ts

```diff
@@ -81,8 +81,14 @@ function createReducerPart(model: any) {
   return new QueryPart(model, def);
 }
 
+// state can also contain the state it was mapped from, ie. "Alerting (NoData)" which means the alert rule is "Alerting" but
```

Comment:
It's more general than that - in the backend we call that second bit the "Reason" and it's not guaranteed to be an alerting state. So I'd say similarly in this comment. 

The state string can also contain a "reason" for the state like "NoData". Only the first word in the state string affects the display of the state.

Or something like that.